plugins/system_upgrade.py

```diff
@@ -557,6 +567,12 @@ def transaction_download(self):
             state.distro_sync = self.opts.distro_sync
             state.allow_erasing = self.cli.demands.allow_erasing
             state.gpgcheck = self.base.conf.gpgcheck
+            state.gpgcheck_repos = [
+                repo.id for repo in self.base.repos.values()
+                if repo._config.gpgcheck().getValue()]
```

Comment:
You can use `if repo.gpgcheck` instead of `if repo._config.gpgcheck().getValue()`